src/mixins/editMixin.js

```diff
@@ -7,27 +7,24 @@ export default {
     value: { required: true },
   },
   data () {
-    const source = this.value.id ? this.value.__unenriched : this.value
     return {
-      source,
-      edit: cloneDeep(source),
+      edit: cloneDeep(this.value),
     }
   },
   watch: {
-    'value.__unenriched' (current, previous) {
+    'value' (current, previous) {
       // we want to make sure it's _really_ changed or we risk undoing the users changes
       if (current !== previous || !deepEqual(current, previous)) {
```

Comment:
We have to be careful here as enriched fields might change based on other data, and if it changed then it would cause the form to be reset and overwrite the users changes.

The reason to have the watching in the first place is to get the updated data after saving to the server (other fields could have changed (or an id added in the case of creation)).

One solution would be to store a changes object, so we can reset from `value`, then reapply any changes (where there should be none in the case of updating due to save, and some in the case of update due to enriched field change).

Although I would be ok if we don't address this now, but to make a note of it somewhere (docs, or an issue).

Reply:
You're right, this should be addressed. As I see it, this would mean a change to `editMixin` only, so I'd do it in a later PR.

What would be a case where this behavior would get visible to the user? I can't think of any right now, it should only trigger during page load and after submit.

Reply:
See #771